upstairs/src/lib.rs

```diff
@@ -2407,22 +2445,126 @@ fn show_guest_work(guest: &Arc<Guest>) {
     println!("GW_JOB completed count:{:?} ", done.len());
 }
 
+/*
+ * IO operations are ok to submit directly to Upstairs if:
+ *
+ * - the offset is block aligned, and
+ * - the size is a multiple of block size
+ *
+ * If either of these is not true, then perform some fixup here.
+ */
+#[derive(Debug)]
+struct IOSpan {
+    // IOP details
+    offset: u64,
+    sz: u64,
+
+    // Block and Guest details
+    block_size: u64,
+    phase: u64,
+    buffer: Buffer,
+    affected_block_numbers: Vec<u64>,
+}
+
+impl IOSpan {
+    // Create an IOSpan given a IO operation at offset and size.
+    fn new(offset: u64, sz: u64, block_size: u64) -> IOSpan {
+        let start_block = offset / block_size;
+        let end_block = (offset + sz - 1) / block_size;
+
+        let affected_block_numbers: Vec<u64> =
+            (start_block..=end_block).collect();
+
+        Self {
+            offset,
+            sz,
+            block_size,
+            phase: offset % block_size,
+            buffer: Buffer::new(
+                affected_block_numbers.len() * block_size as usize,
+            ),
+            affected_block_numbers,
+        }
+    }
+
+    fn is_block_aligned_and_block_sized(&self) -> bool {
```

Comment:
This, and a few of the other function names here I think might be too long.  What if it was named span_check_alignment, or span_alignment, or span_align_check?  Or maybe span_align_ok?

Reply:
:+1: I named it `is_block_regular` in 92eb012, but I'm not sure how to make the other function names shorter.